### PR TITLE
Add a Default app template

### LINKED ISSUE #841: Components should have some pre-filled dummy data when dragged on canvas
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

This would assist in getting started with the components. Component names:
1. Data grid
2. Select
3. Image

Also, for Select and Textfield components Label="Label" should be there by default.

### PR DESCRIPTION
**Pull request by bharatkashyap**
- Closes #841 
- Visual demonstration 

https://user-images.githubusercontent.com/19550456/204729174-ac0d1bed-43f6-497e-982a-ae91fcf98c4d.mov

@apedroferreira Does something like this work for the demo as well? This is what I meant in https://github.com/mui/mui-toolpad/pull/1323#issuecomment-1326748742

### CODE REVIEW

**Comment by prakhargupta1** [comment]
It looks good! Is this what we'll show as dummy data in the data grid?
We can definitely extend this as a proper template.

**Comment by bharatkashyap** [comment]
> Is this what we'll show as dummy data in the data grid?

My idea is that we don't add dummy data to every single data grid, and make this template the default. That allows first time users to see dummy data in the data grid, but removed the need for someone to have to remove dummy data from a data grid every successive time.

File: test/models/ToolpadHome.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export class ToolpadHome {
     });
     this.newAppNameInput = this.newAppDialog.locator('label:has-text("name")');
     this.newAppTemplateSelect = this.newAppDialog.locator(
-      '[aria-haspopup="listbox"]:has-text("blank")',
+      '[aria-haspopup="listbox"]:has-text("demo")',
```
Comment (apedroferreira) [inline test/models/ToolpadHome.ts:43]: I know this wasn't correct before the PR already, but we should try to select this with the `Use template` label if there's a way.

**Review by apedroferreira — APPROVED** [review]
LGTM if we want this template to be the new default one.
I added a comment about trying to make the test not rely on the default app template, but could be fixed separately too, I could try it.

File: test/models/ToolpadHome.ts (hunk 1 above)
Comment (bharatkashyap) [reply test/models/ToolpadHome.ts:43]: How would we select the option with the `Use template` label? (if I'm understanding what you're saying correctly)

Comment (Janpot) [reply test/models/ToolpadHome.ts]: https://github.com/mui/mui-toolpad/blob/28a36db057e184243941db39d7a1ef45130aaa76/test/integration/appTemplates.spec.ts#L33-L35

Comment (apedroferreira) [reply test/models/ToolpadHome.ts]: > How would we select the option with the Use template label?
It's not the label for one of the options, it's the label for the whole select input.

Comment (bharatkashyap) [reply test/models/ToolpadHome.ts]: I've done it with the method @Janpot mentioned up there, seems to work

**Comment by bharatkashyap** [comment]
> @bharatkashyap Since we're trying to rely on external datasources as little as possible to avoid issues that we can't control, are the images in this template safe to use, or should we use our own images in the `static` folder? If so, I guess that 5 or 6 small royalty-free photos for example should work?

I've used data from https://mockapi.io which has `https://cloudflare-ipfs.com` links to the static assets, I feel that is a relatively safe choice for default data?

**Pull request merged**